src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -16,15 +21,33 @@
             + "Example: " + COMMAND_WORD + " alice bob charlie";
 
     private final Set<String> keywords;
+    private final Deadline deadline;
+
+    public FindCommand(String keywords, String deadline)
+            throws IllegalValueException {
+        final Set<String> keywordSet = new HashSet<>();
+        StringTokenizer st = new StringTokenizer(keywords, " ");
+        while (st.hasMoreTokens()) {
+            keywordSet.add(st.nextToken());
+        }
+        this.keywords = keywordSet;
+        this.deadline = createDeadline(deadline);
 
-    public FindCommand(Set<String> keywords) {
-        this.keywords = keywords;
     }
 
     @Override
     public CommandResult execute() {
-        model.updateFilteredTaskList(keywords);
-        return new CommandResult(getMessageForTaskListShownSummary(model.getFilteredTaskList().size()));
+        if (!keywords.equals("")) {
+            model.updateFilteredTaskListByName(keywords);
+            return new CommandResult(getMessageForTaskListShownSummary(model.getFilteredTaskList().size()));
+        } else {
+            model.updateFilteredTaskListByDate(deadline);
+            return new CommandResult(getMessageForTaskListShownSummary(model.getFilteredTaskList().size()));
+        }
+    }
+
+    public Deadline createDeadline(String deadline) throws IllegalValueException {
```

Comment:
Refer to `FindCommandParser`. `deadline` is default to be `""` if not provided by user.